src/alert/alert.component.ts

```diff
@@ -1,23 +1,28 @@
-import {
-  ChangeDetectionStrategy, ChangeDetectorRef, Component, EventEmitter, Input, OnInit,
-  Output
-} from '@angular/core';
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
 import { AlertConfig } from './alert.config';
-import { OnChange } from '../utils/decorators';
+
+export type AlertType = 'success' | 'info' | 'warning' | 'danger';
 
 @Component({
   selector: 'alert,bs-alert',
   templateUrl: './alert.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class AlertComponent implements OnInit {
-  /** Alert type.
-   * Provides one of four bootstrap supported contextual classes:
-   * `success`, `info`, `warning` and `danger`
-   */
-  @Input() type = 'warning';
-  /** If set, displays an inline "Close" button */
-  @OnChange()   @Input()   dismissible = false;
+  /** Alert type */
+  @Input() type: AlertType = 'warning';
+
+  get dismissible(): boolean {
+    return this._dismissible;
+  }
+
+  /** If set, displays an inline 'Close' button */
+  @Input()
+  set dismissible(value: boolean) {
+    this._dismissible = value;
+    this.classes = this.dismissible ? 'alert-dismissible' : '';
```

Comment:
if alert will be used inside of OnPush component, this change will take effect without call to markForCheck

Reply:
I'm not sure I understand your case. This should work for every scenario.

Reply:
try this demo locally https://valor-software.com/ngx-bootstrap/#/alerts#dismissing

Reply:
you don't have to close it :)
I have no rush for it

Reply:
👌